### PR TITLE
util-linux: Depends on python on Linux

### PR DESCRIPTION
**Pull request by dholm**

### CODE REVIEW

**Comment by sjackman** [comment]
What's the build error that you trying to resolve this this PR? Please report the output of `brew install -s util-linux && brew gist-logs util-linux` without this PR applied.

**Comment by dholm** [comment]
`brew gist-logs util-linux` just hangs when I call it but here is the relevant build error:

```
/bin/sh ./libtool  --tag=CC   --mode=compile gcc-5 -DHAVE_CONFIG_H -I.  -include config.h -I./include -DLOCALEDIR=\"/home/eholmda/.linuxbrew/Cellar/util-linux/2.32.1_1/share/locale\" -D_PATH_RUNSTATEDIR=\"/home/eholmda/.linuxbrew/Cellar/util-linux/2.32.1_1/var/run\"  -fsigned-char -fno-common -Wall -Werror=sequence-p
oint -Wextra -Wmissing-declarations -Wmissing-parameter-type -Wmissing-prototypes -Wno-missing-field-initializers -Wredundant-decls -Wsign-compare -Wtype-limits -Wuninitialized -Wunused-but-set-parameter -Wunused-but-set-variable -Wunused-parameter -Wunused-result -Wunused-variable -Wnested-externs -Wpointer-arith -W
strict-prototypes -Wimplicit-function-declaration  -I./libblkid/src -I./libmount/src -I./libmount/src -g -O2 -c -o libmount/src/libmount_la-context_umount.lo `test -f 'libmount/src/context_umount.c' || echo './'`libmount/src/context_umount.c
In file included from libmount/python/fs.c:28:0:                                                                                                                                                                                                                                                                              
libmount/python/pylibmount.h:4:20: fatal error: Python.h: No such file or directory                                                                                                                                                                                                                                           
compilation terminated.                                                                                                                                              
make[2]: *** [Makefile:9293: libmount/python/pylibmount_la-fs.lo] Error 1                                                                                                                                                                                                                                                     
make[2]: *** Waiting for unfinished jobs....                                                                                                                                                                                                                                                                                  
In file included from libmount/python/context.c:21:0:                                                                                                
libmount/python/pylibmount.h:4:20: fatal error: Python.h: No such file or directory                                                                                                                                                                                                                                           
compilation terminated.                                                                                                                                                                                                                                                                                                       
In file included from libmount/python/pylibmount.c:21:0:                                                                                                                                                                                                                                                                     
libmount/python/pylibmount.h:4:20: fatal error: Python.h: No such file or directory                                                                                                                                                                                                                                           
compilation terminated.                                                                                                                                                                                                                                                                                                       
make[2]: *** [Makefile:9307: libmount/python/pylibmount_la-context.lo] Error 1                                                                             
make[2]: *** [Makefile:9286: libmount/python/pylibmount_la-pylibmount.lo] Error 1                                                                                                                                                                                                                                             
In file included from libmount/python/tab.c:21:0:                                                                                                                                                                                                                                                                             
libmount/python/pylibmount.h:4:20: fatal error: Python.h: No such file or directory                                                                                
compilation terminated.                                                                                                                                                                                                                                                                                                       
make[2]: *** [Makefile:9300: libmount/python/pylibmount_la-tab.lo] Error 1
```

**Comment by iMichka** [comment]
> Does configure have a --without-python option? If so, what functionality is lost?

On mac it probably uses system python on out the box. We should depend on `python@2` here to be consistent with upstream.

File: Formula/util-linux.rb [hunk 1]
```diff
@@ -14,6 +14,7 @@ class UtilLinux < Formula
     sha256 "796aaa86a98f50fbaf44e357b54f69284a5923290ef8ea3c09434504ae4a3731" => :x86_64_linux
   end
 
+  depends_on "python" unless OS.mac?
```
Comment (iMichka) [inline Formula/util-linux.rb]: Please put this in a single `unless OS.mac? ... end` block:
> depends_on "python"
> depends_on "ncurses"
> depends_on "linuxbrew/extra/linux-pam" => :optional

**Comment by sjackman** [comment]
> On mac it probably uses system python on out the box. We should depend on python@2 here to be consistent with upstream.

They use `python@2` because that's what's provided by `/usr/bin/python`. If `util-linux` works with `python` 3, I'm okay with using that instead. In this case, I don't feel the consistency is needed.

**Comment by iMichka** [comment]
I tried to enforce this consistency when possible. Just to avoid discussions about why that one tool uses Python vs Python 3. It is also easier to track in case this repo needs to be merged with upstream, as we have been consistent with them (though that may not happen before 2020 and the EOL of Python 2).

**Comment by sjackman** [comment]
Okay. Fine by me.

File: Formula/util-linux.rb (hunk 1 above)
Comment (dholm) [reply Formula/util-linux.rb]: @iMichka Fixed!

**Comment by dholm** [comment]
@iMichka Changed to `python@2`.

File: Formula/util-linux.rb [hunk 2]
```diff
@@ -38,6 +41,7 @@ def install
       args << "--disable-cal" # Conflicts with bsdmainutils
       args << "--without-systemd" # Do not install systemd files
       args << "--with-bashcompletiondir=#{bash_completion}"
+      args << "--with-python=3"
```
Comment (sjackman) [inline Formula/util-linux.rb]: Please remove this line.

File: Formula/util-linux.rb [hunk 3]
```diff
@@ -14,8 +14,11 @@ class UtilLinux < Formula
     sha256 "e9b848c71aa03ed81c425e4d73d9bdd5e955cb496aea0edc37753bb8282c21ba" => :x86_64_linux
   end
 
-  depends_on "ncurses" unless OS.mac?
-  depends_on "linuxbrew/extra/linux-pam" => :optional
+  unless OS.mac?
+    depends_on "python@2"
+    depends_on "ncurses"
```
Comment (sjackman) [inline Formula/util-linux.rb]: Please list dependencies in alphabetical order.

**Comment by maxim-belkin** [comment]
@sjackman I pushed the changes you requested.

As a side note: shall we also enable the `--disable-rpath` option? The `./configure --help` says:
```
--disable-rpath         do not hardcode runtime library paths
```

**Comment by iMichka** [comment]
Thanks @dholm !

**Pull request merged**